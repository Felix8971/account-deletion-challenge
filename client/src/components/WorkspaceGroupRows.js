import { map } from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';

export const WorkspaceGroupRows = props =>
  !props.shouldDisplay ? null : (
    <div>
      <h3>{props.groupTitle}</h3>
      <div>
        { map(props.workspaces, workspace => (
          <div key={workspace.spaceId} style={{ marginTop: '1rem' }}>
            <div>Workspace: {workspace.displayName}</div>
            <div>
              {React.Children.count(props.children) === 0
                ? null
                : React.cloneElement(props.children, { workspace })}
            </div>
          </div>
        ))}
      </div>
    </div>
  )

WorkspaceGroupRows.propTypes = {
  groupTitle: PropTypes.string,
  workspaces: PropTypes.array.isRequired,
  children: PropTypes.node,
  shouldDisplay: PropTypes.bool,
}

export default WorkspaceGroupRows;