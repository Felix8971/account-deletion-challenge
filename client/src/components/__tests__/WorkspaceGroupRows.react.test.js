import React from 'react';
import WorkspaceGroupRows from '../WorkspaceGroupRows';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
configure({ adapter: new Adapter() });

const requiredTransferWorkspaces = [
  {
    spaceId: 'workspace1',
    displayName: 'Lightning strike',
    transferableMembers: [
      {
        _id: 'user2',
        name: 'Ryan Lynch',
      },
    ],
  },
  {
    spaceId: 'workspace2',
    displayName: 'Time machine',
    transferableMembers: [
      {
        _id: 'user5',
        name: 'Edward Bayer',
        workspaceId: 'workspace3',
      },
    ],
  },
];

describe('WorkspaceGroupRows', () => {

  it('should be defined', () => {
    expect(WorkspaceGroupRows).toBeDefined();
  });

  it('should render correctly', () => {
    const tree = shallow(
      <WorkspaceGroupRows
        shouldDisplay={true}
        groupTitle="Title"
        workspaces={requiredTransferWorkspaces}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should contain an h3 tag', () => {
    const tree = shallow(
      <WorkspaceGroupRows
        shouldDisplay={true}
        groupTitle="Title"
        workspaces={requiredTransferWorkspaces}
      />
    );
    expect(tree.find('h3')).toHaveLength(1);
  });
});
