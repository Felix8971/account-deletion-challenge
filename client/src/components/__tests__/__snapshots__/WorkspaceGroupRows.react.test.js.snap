// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkspaceGroupRows should render correctly 1`] = `
<div>
  <h3>
    Title
  </h3>
  <div>
    <div
      key="workspace1"
      style={
        Object {
          "marginTop": "1rem",
        }
      }
    >
      <span>
        Workspace: 
        Lightning strike
      </span>
      <span />
    </div>
    <div
      key="workspace2"
      style={
        Object {
          "marginTop": "1rem",
        }
      }
    >
      <span>
        Workspace: 
        Time machine
      </span>
      <span />
    </div>
  </div>
</div>
`;
